CLI/src/config/skeletons.xml

```diff
@@ -172,77 +172,83 @@
   </skeleton>
 
   <!-- replace DECLARE FUNCTION PUBLIC by a stacked program -->
-  <skeleton name="FUN.DECLARE.PUBLIC" var="definitions,copyname,programName8">
+  <skeleton name="FUN.DECLARE.PUBLIC" var="definitions,copyname,programName8,maxargscount">
     <conditions>
       <condition node="TypeCobol.Compiler.Nodes.FunctionDeclaration" visibility="public" isnested="false"/>
     </conditions>
     <patterns>      
-      <pattern location="program.data-division.working-storage" action="create" group="TCRFUN_CODEGEN_IS_LOADED">
-01  TC-@Model.programName8-FctList-Loaded PIC X(02).
-    88 TC-@Model.programName8-FctList-IsLoaded      VALUE 'OK'.
+     <pattern location="program.data-division.working-storage" action="create" newline="false" group="TCRFUN_CODEGEN_IS_LOADED">
       </pattern>
-      <!-- CALLEE WORKING-STORAGE SECTION WITHOUT EXTERNAL -->
-      <pattern location="program.data-division.working-storage" action="create" group="TCRFUN_CODEGEN_POINTER_ARRAY" deprecated="false">
+      <!-- CALLEE LINKAGE SECTION WITHOUT EXTERNAL -->
+      <pattern location="program.data-division.linkage" action="create" newline="false" group="TCRFUN_CODEGEN_POINTER_LINKAGE" deprecated="false">
         @{
-        var items = "";
-        if (@Model.definitions.functions.Public.Count > 0) {
-        items += "01 TC-"+ @Model.programName8 + "-PntTab.\n";
-        items += "    05 TC-" + @Model.programName8 + "-PntNbr         PIC S9(04) COMP VALUE "+@Model.definitions.functions.Public.Count+".\n";
-        }
+        var items = "01 TC-FunctionCode pic X(30).\n";
         foreach (var f in @Model.definitions.functions.Public) {
-        items += "*To call program " + f.Hash + '\n';
-        items += "*Which is generated code for " + f.QualifiedName + '\n';
-        items += "*Declared in source file " + f.CodeElement.TokenSource.SourceName + '\n';
-        items += "    05 TC-"+@Model.programName8 + "-" + f.Hash +"-Idt   PIC X(08) VALUE '" + f.Hash + "'.\n";
-        items += "    05 TC-"+@Model.programName8 + "-" + f.Hash +" PROCEDURE-POINTER.\n";
+        items += "* Function which call program " + f.Hash + "\n";
+        items += "* Which is generated code for " + f.QualifiedName + "\n";
+        items += "    88 Fct-" + f.Hash + "-" + f.ID + "\n";
+        items += "       value 'Fct=" + f.Hash + "-" + f.ID + "'.\n";
         }
+        items += "\n";
+        for(var i=1; i != (@Model.maxargscount + 1); i++) {
+        items += "01 arg" + i + " pic X.\n";
+        }
+        items = items.Substring(0, items.Length - 1);
         }
         @items
       </pattern>
-      <!-- CALLEE LINKAGE SECTION WITHOUT EXTERNAL -->
-      <pattern location="program.data-division.linkage" action="create" group="TCRFUN_CODEGEN_POINTER_LINKAGE" deprecated="false">
+
+      <pattern action="expand" location="program.end" newline="false">
+      </pattern>
```

Comment:
same as above

Reply:
after check, this pattern is mandatory (else throwed exception).